searchlib/src/vespa/searchlib/attribute/raw_attribute.cpp

```diff
@@ -11,32 +12,58 @@ RawAttribute::RawAttribute(const vespalib::string& name, const Config& config)
 
 RawAttribute::~RawAttribute() = default;
 
-long
-RawAttribute::onSerializeForAscendingSort(DocId doc, void* serTo, long available, const common::BlobConverter*) const
+namespace {
+
+template <bool desc>
+unsigned char remap(unsigned char val)
 {
-    auto raw = get_raw(doc);
-    if (available >= (long)raw.size()) {
-        memcpy(serTo, raw.data(), raw.size());
+    return (desc ? (0xff - val) : val);
+}
+
+template <bool desc>
+long serialize_for_sort(vespalib::ConstArrayRef<char> raw, void* serTo, long available)
```

Comment:
Please add a description of the serialization format, and why the trailing zero is needed.